client/ayon_core/hosts/substancepainter/plugins/create/create_textures.py

```diff
@@ -42,11 +49,20 @@ def create(self, product_name, instance_data, pre_create_data):
             "exportFileFormat",
             "exportSize",
             "exportPadding",
-            "exportDilationDistance"
+            "exportDilationDistance",
+            "useCustomExportPreset",
+            "exportChannel"
         ]:
             if key in pre_create_data:
                 creator_attributes[key] = pre_create_data[key]
 
+        if pre_create_data.get("use_selection"):
+            stack = substance_painter.textureset.get_active_stack()
+
+            instance_data["selected_node_id"] = [
+                node_number.uid() for node_number in
+                substance_painter.layerstack.get_selected_nodes(stack)]
```

Comment:
Do these `uid()` values persist after a restart of substance painter? Since this would be data we're storing with the scene?

Reply:
> Do these `uid()` values persist after a restart of substance painter? Since this would be data we're storing with the scene?

They won't be changed, as they are based on node type.
`[Python] FillLayerNode[220]`